### PR TITLE
Add sorting by name or creation date to the /plugins endpoint

### PR DESCRIPTION
**Pull request by PartyWumpus**
adds `sort_by` and  `sort_direction` to /plugins query with options [`name` and `date`] and [`asc` and `desc`] respectively. No idea if it actually works, as i have zero clue how to test this stuff. (i intend to add more sorting options later)

### CODE REVIEW

File: plugin_store/database/database.py [hunk 1]
```diff
@@ -162,6 +179,15 @@ async def search(
...
             statement = statement.where(Artifact.visible.is_(True))
+        direction = desc()
+        match (order_by):
+            case SortOption.asc: direction = asc()
+            case SortOption.desc: direction = desc()
+            case SortOption.NONE: pass
+        match (sort_by):
+            case SortType.name: statement = statement.order_by(direction(Artifact.name))
+            case SortType.date: statement = statement.order_by(direction(Artifact.created))
+            case SortType.NONE: pass
```
Comment (gbdlin) [inline plugin_store/database/database.py]: We're still using python 3.9, match is not available.

File: plugin_store/api/__init__.py [hunk 2]
```diff
@@ -65,15 +66,28 @@ async def index():
     return INDEX_PAGE
 
 
+# TODO: move these enums somewhere to deduplicate them
+class SortOption(StrEnum):
+    desc = auto()
+    asc = auto()
+    NONE = auto()
```
Comment (gbdlin) [inline plugin_store/api/__init__.py]: Can we just use a default of `desc` in functions using that and get rid of `NONE`? it essentially means `desc` as this is the default hardcoded later.

File: plugin_store/api/__init__.py [hunk 3]
```diff
@@ -65,15 +66,28 @@ async def index():
...
+
+class SortType(StrEnum):
+    name = auto()
+    date = auto()
+    NONE = auto()
```
Comment (gbdlin) [inline plugin_store/api/__init__.py]: can we use `None` do indicate no sorting instead? Then there would be no field for that in this enum, and sort param would be defined as `Optional[SortOption]` instead.

File: plugin_store/api/__init__.py [hunk 4]
```diff
@@ -65,15 +66,28 @@ async def index():
...
+
 @app.get("/plugins", response_model=list[api_list.ListPluginResponse])
 async def plugins_list(
     query: str = "",
     tags: list[str] = fastapi.Query(default=[]),
     hidden: bool = False,
+    sort_by: SortType = SortType.NONE,
+    order_by: SortOption = SortOption.NONE,
```
Comment (gbdlin) [inline plugin_store/api/__init__.py]: what about `sort_direction` and `sort_by`? or `order_direction` or `order_by`?

**Review by gbdlin — COMMENTED** [review]
We need some tests for that as well

File: plugin_store/database/database.py (hunk 1 above)
Comment (PartyWumpus) [reply plugin_store/database/database.py]: That's a shame. I'll switch it to if else then :(

**Comment by PartyWumpus** [comment]
It's working well on my local version, I'll work on adding tests for it tomorrow. (i just did the video cuz i was bored)

https://github.com/SteamDeckHomebrew/decky-plugin-store/assets/48649272/8ee99d6a-36e9-4780-bf97-bb22c4038da5

**Comment by PartyWumpus** [comment]
Added 7 tests to `test_plugins_list_endpoint`, but the total number of tests has gone from 320 to 2048. Should i remove some of the tests or move them somewhere else?

**Review by gbdlin — APPROVED** [review]

**Pull request merged**